### PR TITLE
 fixed non-determinism and errors when indexing NYT collection (#234)

### PR DESCRIPTION
**Pull request by Kytabyte**
As per #234, the bug arises from two parts, one is when handling empty-body documents, and the other is due to a static variable shared across instances of `NYTCorpusDocumentParser` which should not be. Just two simple tweaks to fix the issue.

### CODE REVIEW

File: src/main/java/io/anserini/document/nyt/NYTCorpusDocumentParser.java [hunk 1]
```diff
@@ -65,7 +65,7 @@
 	private static final String SERIES_NAME_TAG = "series.name";
 
 	/** NITF Constant */
-	private static final DateFormat format = new SimpleDateFormat(
+	private DateFormat format = new SimpleDateFormat(
 			"yyyyMMdd'T'HHmmss");
```
Comment (lintool) [inline src/main/java/io/anserini/document/nyt/NYTCorpusDocumentParser.java]: Don't we still want `final`?

Comment (Kytabyte) [reply src/main/java/io/anserini/document/nyt/NYTCorpusDocumentParser.java]: Yes.

**Comment by lintool** [comment]
+ Please update `docs/experiments-core17.md`
+ Add regression test script, see `src/main/python/run_regression_robust04.py`

**Comment by Kytabyte** [comment]
Done!

**Comment by Kytabyte** [comment]
And the script works with the result

```
All Topics: bm25     : map : 0.1996 0.1996
All Topics: bm25+rm3 : map : 0.2534 0.2534
All Topics: ql       : map : 0.1928 0.1928
All Topics: ql+rm3   : map : 0.2363 0.2363
All Topics: bm25     : p30 : 0.4207 0.4207
All Topics: bm25+rm3 : p30 : 0.4767 0.4767
All Topics: ql       : p30 : 0.4327 0.4327
All Topics: ql+rm3   : p30 : 0.4640 0.4640
```

**Pull request merged**